tests/test_pipelines_token_classification.py

```diff
@@ -262,15 +59,273 @@ def _test_pipeline(self, nlp: Pipeline):
+                "entity_group": "PER",
+                "score": 0.9997771680355072,
+                "word": "Andrés Pastrana",
+                "start": 23,
+                "end": 37,
+            },
+            {"entity_group": "ORG", "score": 0.9989739060401917, "word": "Farc", "start": 39, "end": 43},
+        ]
+        self.assertEqual(nlp.aggregate(example1, AggregationStrategy.NONE), example1)
+        self.assertEqual(nlp.aggregate(example1, AggregationStrategy.SIMPLE), simple_result)
+        self.assertEqual(nlp.aggregate(example1, AggregationStrategy.FIRST), first_result)
+        # We don't test MAX and AVERAGE here, the examples output roughly the same
+
+    @require_torch
+    def test_aggregation_strategy_example2(self):
+        model_name = self.small_models[0]
+        tokenizer = AutoTokenizer.from_pretrained(model_name, use_fast=True)
+        nlp = pipeline(task="ner", model=model_name, tokenizer=tokenizer, framework="pt")
+        example2 = [
+            {
+                "entity": "I-PER",
+                "index": 1,
+                "score": 0.9968166351318359,
+                "is_subword": False,
+                "word": "En",
+                "start": 0,
+                "end": 2,
+            },
+            {
+                "entity": "I-PER",
+                "index": 2,
+                "score": 0.9957635998725891,
+                "is_subword": True,
+                "word": "##zo",
+                "start": 2,
+                "end": 4,
+            },
+            {
+                "entity": "I-ORG",
+                "index": 7,
+                "score": 0.9986497163772583,
+                "is_subword": False,
+                "word": "UN",
+                "start": 11,
+                "end": 13,
+            },
+        ]
+        simple_result = [
+            {"entity_group": "PER", "score": 0.9962901175022125, "word": "Enzo", "start": 0, "end": 4},
+            {"entity_group": "ORG", "score": 0.9986497163772583, "word": "UN", "start": 11, "end": 13},
+        ]
+        # Only the score vary.
+        first_result = [
+            {"entity_group": "PER", "score": 0.9968166351318359, "word": "Enzo", "start": 0, "end": 4},
+            {"entity_group": "ORG", "score": 0.9986497163772583, "word": "UN", "start": 11, "end": 13},
+        ]
+        self.assertEqual(nlp.aggregate(example2, AggregationStrategy.NONE), example2)
+        self.assertEqual(nlp.aggregate(example2, AggregationStrategy.SIMPLE), simple_result)
+        self.assertEqual(nlp.aggregate(example2, AggregationStrategy.FIRST), first_result)
+        # We don't test MAX and AVERAGE here, the examples output roughly the same
+
+    @require_torch
+    def test_aggregation_strategy_example3(self):
+        model_name = self.small_models[0]
+        tokenizer = AutoTokenizer.from_pretrained(model_name, use_fast=True)
+        nlp = pipeline(task="ner", model=model_name, tokenizer=tokenizer, framework="pt")
+        example3 = [
```

Comment:
Does this implementation support a scenario where we need to aggregate subwords into words, but not words into entities (e.g. in case the model labels do not follow the BIO convention we don't want to merge multiple entities with the same class)?
I think it doesn't, but I might have missed some details of the implementation logic.

Reply:
Ok, you're right that this currently does not support that, is that correct that in the absence of '-' within the Tag name we can assume everything is 'B-' ?

Reply:
Maybe it's more robust to check explicitly in the label dict if they match the BIO pattern. Using a conditional on absence of '-' seems brittle as it could yield a false negative if a label is e.g. 'over-the-counter' (imagine a model classifying medications in prescription notes).

Reply:
I've chnaged the code to be more restrictive.